src/lib/Libwin/passwd.c

```diff
@@ -3949,15 +3952,19 @@ setuser_with_password(char *user,
 	if ((user == NULL) || (decrypt_func == NULL))
 		return (0);
 
-	/* look in internal array for saved usertoken handle */
-	for (i=0; i<user_num; i++) {
-		pwdp = &user_array[i];
+	if(user_num == 0){
```

Comment:
This variable should not be required. Initialize (CLEAR_HEAD) the linked list head once at startup, and that should we all required.